kong/runloop/balancer/round_robin.lua

```diff
@@ -92,36 +97,51 @@ function roundrobin_algorithm:getPeer(cacheOnly, handle, hashValue)
     end
 
     address = self.wheel[self.pointer]
-    if address ~= nil and address.available and not address.disabled then
-      ip, port, hostname = balancers.getAddressPeer(address, cacheOnly)
-      if ip then
-        -- success, update handle
-        handle.address = address
-        return ip, port, hostname, handle
-
-      elseif port == balancers.errors.ERR_DNS_UPDATED then
-        -- if healty we just need to try again
-        if not self.balancer.healthy then
-          return nil, balancers.errors.ERR_BALANCER_UNHEALTHY
-        end
-      elseif port == balancers.errors.ERR_ADDRESS_UNAVAILABLE then
-        ngx.log(ngx.DEBUG, "found address but it was unavailable. ",
-          " trying next one.")
-      else
-        -- an unknown error occurred
-        return nil, port
+    if not address or not address.available or address.disabled then
```

Comment:
Also, changing the order of if else really doesn't matter much. If you want to make changes, you can standardize the code of several balancer algorithms together so that they follow one style."  https://github.com/Kong/kong/tree/master/kong/runloop/balancer

Reply:
This PR is my first try, if it is accepted and merged I will open other PR.